src/components/Contractor/Profile/useContractorProfile.ts

```diff
@@ -0,0 +1,357 @@
+import { useMemo } from 'react'
+import { zodResolver } from '@hookform/resolvers/zod'
+import type { SubmitHandler } from 'react-hook-form'
+import { useForm, useWatch } from 'react-hook-form'
+import { z } from 'zod'
+import { useContractorsCreateMutation } from '@gusto/embedded-api/react-query/contractorsCreate'
+import { useContractorsUpdateMutation } from '@gusto/embedded-api/react-query/contractorsUpdate'
+import { useContractorsGet } from '@gusto/embedded-api/react-query/contractorsGet'
+import type { PostV1CompaniesCompanyUuidContractorsRequestBody } from '@gusto/embedded-api/models/operations/postv1companiescompanyuuidcontractors'
+import type { PutV1ContractorsContractorUuidRequestBody } from '@gusto/embedded-api/models/operations/putv1contractorscontractoruuid'
+import {
+  WageType as ApiWageType,
+  ContractorType as ApiContractorType,
+} from '@gusto/embedded-api/models/components/contractor'
+import { useBase } from '@/components/Base'
+import { useI18n } from '@/i18n'
+import { componentEvents } from '@/shared/constants'
+import { nameValidation, SSN_REGEX } from '@/helpers/validations'
+import { removeNonDigits } from '@/helpers/formattedStrings'
+import { formatDateToStringDate } from '@/helpers/dateFormatting'
+import { normalizeEin } from '@/helpers/federalEin'
+
+// Re-export the API types for convenience
+export const WageType = ApiWageType
+export const ContractorType = ApiContractorType
+
+// Form schema with conditional validation - let RHF handle it through zodResolver
+export const ContractorProfileSchema = z
+  .object({
+    // Self-onboarding toggle
+    inviteContractor: z.boolean().default(false),
+    email: z.string().email().optional(),
+
+    // Required contractor fields
+    contractorType: z.enum([ContractorType.Individual, ContractorType.Business]),
+    wageType: z.enum([WageType.Hourly, WageType.Fixed]),
+    startDate: z.date(),
+
+    // Individual contractor fields
+    firstName: nameValidation.optional(),
+    middleInitial: z.string().optional(),
+    lastName: nameValidation.optional(),
+    ssn: z.string().optional(),
+
+    // Business contractor fields
+    businessName: z.string().optional(),
+    ein: z.string().optional(),
+
+    // Wage fields
+    hourlyRate: z.number().min(0).optional(),
+  })
+  .superRefine((data, ctx) => {
+    // Email validation for contractor invitation
+    if (data.inviteContractor && !data.email) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        path: ['email'],
+        message: 'Email is required when inviting contractor',
```

Comment:
Are these validation messages actually surfaced to the user? Do we need to add translations for these?

Reply:
They are, I updated them to have translations and also made sure its testabel in storybook.